build/RuntimeStore.targets

```diff
@@ -0,0 +1,176 @@
+    <_SrcDirectory>$(RepositoryRoot)src\</_SrcDirectory>
+    <_AllMetapackageDirectory>$(_SrcDirectory)Microsoft.AspNetCore.All\</_AllMetapackageDirectory>
+    <_AllMetapackageProject>$(_AllMetapackageDirectory)Microsoft.AspNetCore.All.csproj</_AllMetapackageProject>
+    <_ExistingManifestsDirectory>$(_AllMetapackageDirectory)build\</_ExistingManifestsDirectory>
+    <_ArtifactsZipDirectory>$(ArtifactsDir)zip\</_ArtifactsZipDirectory>
+    <_StoreZipDirectory>$(_ArtifactsZipDirectory)rs\</_StoreZipDirectory>
+    <_SymbolsZipDirectory>$(_ArtifactsZipDirectory)symbols\</_SymbolsZipDirectory>
+    <_DepsOutputDirectory>$(ArtifactsDir)deps\</_DepsOutputDirectory>
+  </PropertyGroup>
+
+  <Target Name="BuildAllMetapackage" DependsOnTargets="ResolveRepoInfo">
+    <!-- Generate RS.Reference project -->
+    <RepoTasks.AddMetapackageReferences
+      ReferencePackagePath="$(_AllMetapackageProject)"
+      BuildArtifacts="@(ArtifactInfo)"
+      PackageArtifacts="@(PackageArtifact)" />
+
+    <!-- Restore and pack-->
+    <MSBuild Projects="$(_AllMetapackageProject)"
+      Targets="Restore;Pack"
+      Properties="Configuration=Release" />
+  </Target>
+
+  <Target Name="UpdateNuGetConfig">
+    <UpdatePackageSource
+      NuGetConfigPath="$(RepositoryRoot)NuGet.config"
+      SourceName="Dependencies"
+      SourceUri="$(_DependencyBuildDirectory)"
+      Condition="Exists('$(_DependencyBuildDirectory)')" />
+  </Target>
+
+  <Target Name="_PrepareRuntimeStoreBuildAssets">
+    <!-- Clear working directory -->
+    <RemoveDir Directories="$(_WorkRoot)" />
+
+    <!-- Copy and update build assets -->
+    <Copy SourceFiles="$(RepositoryRoot)NuGet.config" DestinationFiles="$(_WorkRoot)NuGet.config" />
+    <Copy SourceFiles="$(_TemplatesDirectory)RS.Manifest\RS.Manifest.csproj" DestinationFiles="$(_WorkRoot)RS.Manifest.csproj" />
+  </Target>
+
+  <Target Name="_ResolveRuntimeStoreRID">
+    <Error Text="Please specify the platform for the runtime package store via RUNTIMESTORE_PLATFORM: {x64|x86}." Condition="'$(RUNTIMESTORE_PLATFORM)' == ''"/>
+
+    <GetOSPlatform>
+      <!-- Returns {Linux, macOS, Windows} -->
+      <Output TaskParameter="PlatformName" PropertyName="OSPlatform" />
+    </GetOSPlatform>
+
+    <PropertyGroup>
+      <RID Condition="'$(OSPlatform)' == 'Windows'">win7-$(RUNTIMESTORE_PLATFORM)</RID>
+      <RID Condition="'$(OSPlatform)' == 'Linux'">linux-$(RUNTIMESTORE_PLATFORM)</RID>
+      <RID Condition="'$(OSPlatform)' == 'macOS'">osx-$(RUNTIMESTORE_PLATFORM)</RID>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="BuildRuntimeStore" DependsOnTargets="UpdateNuGetConfig;_ResolveRuntimeStoreRID;_PrepareRuntimeStoreBuildAssets;_ResolveCurrentSharedFrameworkVersion">
+    <!-- Build runtime store -->
+    <Exec Command="dotnet store -m $(_WorkRoot)RS.Manifest.csproj -f netcoreapp2.0 -r $(RID) -o $(_RuntimeStoreOutputDirectory) --framework-version 2.0.2-* -w $(_RuntimeStoreWorkDirectory) $(RUNTIMESTORE_ADDITIONAL_ARGS)" />
```

Comment:
What would it take to remove the floating version? Floating dependency versions is evil.

Reply:
Good call, should be able to use `$(SharedFrameworkVersion)` now that I parse it. It was hard coded due to limitations that is now ancient history.